src/zenml/orchestrators/utils.py

```diff
@@ -305,3 +308,95 @@ def _get_model_versions_from_artifacts(
             else:
                 break
     return models
+
+
+class artifact_store_handler:
+    """Context manager for the artifact_store/filesystem_registry dependency.
+
+    Even though it is rare, sometimes we bump into cases where we are trying to
+    load artifacts that belong to an artifact store which is different from
+    the active artifact store.
+
+    In cases like this, we will try to instantiate the target artifact store
+    by creating the corresponding artifact store Python object, which ends up
+    registering the right filesystem in the filesystem registry.
+
+    The problem is, the keys in the filesystem registry are schemes (such as
+    "s3://" or "gcs://"). If we have two artifact stores with the same set of
+    supported schemes, we might end up overwriting the filesystem that belongs
+    to the active artifact store (and its authentication). That's why we have
+    to re-instantiate the active artifact store again, so the correct filesystem
+    will be restored.
+    """
+
+    def __init__(self, target_artifact_store_id: Optional[UUID]) -> None:
+        """Initialization of the context manager.
+
+        Args:
+            target_artifact_store_id: the ID of the artifact store to load.
+        """
+        self.active_artifact_store_id = Client().active_stack.artifact_store.id
+        self.target_artifact_store_id = target_artifact_store_id
+
+    def __enter__(self) -> "BaseArtifactStore":
+        """Instantiate the target artifact store and return it.
+
+        Returns:
+            the python object
+        """
+        try:
+            if self.target_artifact_store_id is not None:
+                if (
+                    self.active_artifact_store_id
+                    != self.target_artifact_store_id
+                ):
+                    get_logger(__name__).warning(
```

Comment:
Why do we need a warning for this? Isn't this the whole purpose of this context manager to do exactly that?

Reply:
Maybe, using a warning here is a bit of an overkill. However, since this is a rare occasion that might lead to some interesting behavior, I would at least throw a debug message here.